azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/AbstractConfigParser.java

```diff
@@ -88,10 +91,19 @@ public WebAppConfig getWebAppConfig() throws AzureExecutionException {
                 .build();
     }
 
+    protected MavenDockerCredentialProvider getDockerCredential(String serverId) {
+        return MavenDockerCredentialProvider.fromMavenSettings(mojo.getSettings(), serverId);
+    }
+
     protected void validate(Supplier<String> validator) throws AzureExecutionException {
         final String message = validator.get();
         if (StringUtils.isNotEmpty(message)) {
             throw new AzureExecutionException(message);
         }
     }
+
+    protected DeployType getDeployTypeFromFile(File file) {
+        final DeployType type = DeployType.fromString(FilenameUtils.getExtension(file.getName()));
```

Comment:
IMO. no need to add `String` into the method name. 
java is a strong type language, the user know he/she need to pass a string to the method from the type of parameter.